pkg/dackbox/indexer/lazy.go

```diff
@@ -83,6 +91,23 @@ func (li *lazyImpl) runIndexing() {
 	}
 }
 
+func (li *lazyImpl) isDeduped(doc *document.Document) bool {
```

Comment:
Nit:  the naming is off. `isDuplicate` would be more adequate, but `is...` also suggests a side-effect freeness the this medic doesn't provide. Don't have a good idea off the top of my head on his to name this instead though

Reply:
Yeah, agreed. Let me think of a better name because I don't like the fact that this function modifies something. I should also note that the hashing of the doc itself is a bit of a tradeoff in terms of increasing the likelihood of deduping while potentially paying the cost of doc mapping (which isn't free). However, the true solution would be to hash fields with search labels that aren't `-`, but thats more involved then what I want to do here